newsfragments/40936.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix ``on_success_callback`` will not run if task is skipped.
```

Comment:
We need more details here. It doesnt clarify why this is a breaking change and the consideration.

Reply:
changed it, let me know what you think